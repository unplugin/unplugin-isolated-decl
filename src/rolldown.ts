/**
 * This entry file is for Rolldown plugin.
 *
 * @module
 */

import { IsolatedDecl } from './index.ts'

/**
 * Rolldown plugin
 *
 * @example
 * ```ts
 * // rolldown.config.js
 * import IsolatedDecl from 'unplugin-isolated-decl/rolldown'
 *
 * export default {
 *   plugins: [IsolatedDecl()],
 * }
 * ```
 */
const rolldown = IsolatedDecl.rolldown as typeof IsolatedDecl.rolldown
export default rolldown
export { rolldown as 'module.exports' }
