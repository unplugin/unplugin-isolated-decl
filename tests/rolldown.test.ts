import path from 'node:path'
import { rolldown } from 'rolldown'
import { expect, test } from 'vitest'
import UnpluginIsolatedDecl from '../src/rolldown'

test('rolldown', async () => {
  const input = path.resolve(__dirname, 'fixtures/basic/main.ts')
  const dist = path.resolve(__dirname, 'temp/rolldown/basic')

  const bundle = await rolldown({
    input,
    plugins: [
      UnpluginIsolatedDecl({
        extraOutdir: 'temp',
        transformer: 'oxc',
      }),
    ],
    logLevel: 'silent',
  })

  const result = await bundle.generate({
    dir: dist,
  })
  expect(
    result.output
      .sort((a, b) => a.fileName.localeCompare(b.fileName))
      .map(
        (asset) =>
          `// ${asset.fileName.replaceAll('\\', '/')}\n${asset.type === 'chunk' ? asset.code : asset.source}`,
      ),
  ).toMatchSnapshot()
})
