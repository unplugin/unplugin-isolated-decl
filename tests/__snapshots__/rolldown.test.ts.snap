// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`rolldown 1`] = `
[
  "// main.js
import { jsx as _jsx } from "react/jsx-runtime";

//#region tests/fixtures/component.tsx
function Component() {
	return _jsx("div", { children: "I'm a div in a tsx component!" });
}

//#endregion
//#region tests/fixtures/main.ts
function hello(s) {
	return "hello" + s;
}
let c = Component;
let num = 1;

//#endregion
export { c, hello, num };",
  "// temp/component.d.ts
export declare function Component(): React.JSX.Element;
",
  "// temp/main.d.ts
import { type Num } from "./types";
export type Str = string;
export declare function hello(s: Str): Str;
export declare let c: React.JSX.Element;
export declare let num: Num;
",
  "// temp/types.d.ts
export type Num = number;
",
  "// temp/types2.d.ts
export type Num2 = number;
",
]
`;
